packages/grafana-data/src/types/datasource.ts

```diff
@@ -323,6 +323,11 @@ abstract class DataSourceApi<
    */
   modifyQuery?(query: TQuery, action: QueryFixAction): TQuery;
 
+  /**
+   * Used in explore for Log details
+   */
+  inspectQuery?(query: TQuery, options: InspectQueryOptions): boolean;
```

Comment:
It's a new API method so I suggest marking it as `@alpha` or even `@internal` until we want other plugins to start using it. 

Also could we follow the pattern where a separate interface is introduced? Like `DataSourceWithSupplementaryQueriesSupport` or `DataSourceWithQueryImportSupport`.  Looks like it'd be great if plugins implemented `modifyQuery` with `inspectQuery`, this could be grouped in a separate file like in packages/grafana-data/src/types/query.ts

Also - do we consider this API to be extended in the future and return non-boolean values?

Reply:
> It's a new API method so I suggest marking it as @alpha or even @internal until we want other plugins to start using it.

Good idea. Done.

> Also could we follow the pattern where a separate interface is introduced? 

Definitely. Would it be okay to address it in a follow-up PR?

> Also - do we consider this API to be extended in the future and return non-boolean values?

Absolutely. I thought about it but couldn't find a way that I liked to express it in the return type.

Reply:
> Also - do we consider this API to be extended in the future and return non-boolean values?

Added support to the interface to customize the return type.